sw/airborne/modules/pose_history/pose_history.c

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (C) Roland Meertens
+ *
+ * This file is part of paparazzi
+ *
+ * paparazzi is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+/**
+ * @file "modules/pose_history/pose_history.c"
+ * @author Roland Meertens
+ * Ask this module for the pose the drone had closest to a given timestamp
+ */
+
+#include "modules/pose_history/pose_history.h"
+#include <sys/time.h>
+#include "mcu_periph/sys_time.h"
+#include "state.h"
+#if LINUX
+#include <pthread.h>
+#endif
+
+#ifndef POSE_HISTORY_SIZE
+#define POSE_HISTORY_SIZE 1024
+#endif
+
+typedef struct {
+  uint32_t ring_index;
+  uint32_t ring_size;
+  timeAndRotation ring_data[POSE_HISTORY_SIZE];
+} rotationHistoryRingBuffer;
+
+rotationHistoryRingBuffer location_history;
+
+#if LINUX
+pthread_mutex_t pose_mutex;
+#endif
+
+
+
+void float_eulers_copy(struct FloatEulers *to, struct FloatEulers *from);
+void float_eulers_copy(struct FloatEulers *to, struct FloatEulers *from)
+{
+  to->phi = from->phi;
+  to->theta = from->theta;
+  to->psi = from->psi;
+}
+
+void float_rates_copy(struct FloatRates *to, struct FloatRates *from);
+void float_rates_copy(struct FloatRates *to, struct FloatRates *from)
```

Comment:
If you copy full structs of the same type, you should simply use a normal assignment...